provisioning/device/devdoc/transport_state_machine_requirements.md

```diff
@@ -0,0 +1,81 @@
+# transport_state_machine Requirements
+
+## Overview
+This module provides HTTP protocol support to communicate with the Azure device provisioning service
+
+## Example Usage
+``js
+  // this is effectively an abstract class that transport implements are built upon.  It is not meant to be used directly.
+``
+
+## Public Interface
+
+### constructor()
+The constructor creates a TransportStateMachine object
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_001: [** The `constructor` shall accept no arguments **]**
+
+
+### connect(callback: (err?: Error) => void): void
+Connect is used to connect the transport
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_002: [** `connect` shall call `_doConnectForFsm`. **]**
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_003: [** If `_doConnectForFsm` fails, then `connect` shall fail. **]**
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_004: [** If the transport is already connected, then `connect` shall do nothing and call the `callback` immediately. **]**
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_005: [** If `connect` is called while a connection is in progress, it shall wait for that connection to complete before calling `callback`. **]**
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_006: [** If `connect` is called while the transport is executing the first registration request, it shall do nothing and call `callback` immediately. **]**
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_007: [** If `connect` is called while the transport is waiting between operation status queries, it shall do nothing and call `callback` immediately. **]**
```

Comment:
not possible to disconnect and reconnect between operation status queries?

Reply:
Nope.  status queries come every 2 seconds.  I don't want to open that hole.

---
In reply to: [146396612](https://github.com/Azure/azure-iot-sdk-node/pull/115#discussion_r146396612) [](ancestors = 146396612)